lib/custom/src/MShop/Service/Provider/Payment/Mollie.php

```diff
@@ -47,4 +47,250 @@ protected function getValue( $key, $default = null )
+				$status = \Aimeos\MShop\Order\Item\Base::PAY_RECEIVED;
+			}
+			else
+			{
+				return $order;
+			}
+
+			if( method_exists( $response, 'isSuccessful' ) && $response->isSuccessful() )
+			{
+				$order->setPaymentStatus( $status );
+			}
+			elseif( method_exists( $response, 'isAuthorized' ) && $response->isAuthorized() )
+			{
+				$order->setPaymentStatus( \Aimeos\MShop\Order\Item\Base::PAY_AUTHORIZED );
+			}
+			elseif( method_exists( $response, 'isPending' ) && $response->isPending() )
+			{
+				$order->setPaymentStatus( \Aimeos\MShop\Order\Item\Base::PAY_PENDING );
+			}
+			elseif( method_exists( $response, 'isOpen' ) && $response->isOpen() )
+			{
+				$order->setPaymentStatus( \Aimeos\MShop\Order\Item\Base::PAY_REFUSED );
+			}
+			elseif( method_exists( $response, 'isCancelled' ) && $response->isCancelled() )
+			{
+				$order->setPaymentStatus( \Aimeos\MShop\Order\Item\Base::PAY_CANCELED );
+			}
+			elseif( method_exists( $response, 'isRedirect' ) && $response->isRedirect() )
+			{
+				$url = $response->getRedirectUrl();
+				throw new \Aimeos\MShop\Service\Exception( sprintf( 'Unexpected redirect: %1$s', $url ) );
+			}
+			else
+			{
+				$this->saveOrder( $order->setPaymentStatus( \Aimeos\MShop\Order\Item\Base::PAY_REFUSED ) );
+				throw new \Aimeos\MShop\Service\Exception( $response->getMessage() );
+			}
+
+			$this->saveTransationRef( $base, $response->getTransactionReference() );
+			$this->saveRepayData( $response, $base->getCustomerId() );
+			$this->saveOrder( $order );
+		}
+		catch( \Exception $e )
+		{
+			throw new \Aimeos\MShop\Service\Exception( $e->getMessage() );
+		}
+
+		return $order;
+	}
+	
+	/**
+	 * Tries to get an authorization or captures the money immediately for the given order if capturing the money
+	 * separately isn't supported or not configured by the shop owner.
+	 *
+	 * @param \Aimeos\MShop\Order\Item\Iface $order Order invoice object
+	 * @param array $params Request parameter if available
+	 * @return \Aimeos\MShop\Common\Item\Helper\Form\Standard Form object with URL, action and parameters to redirect to
+	 * 	(e.g. to an external server of the payment provider or to a local success page)
+	 */
+	protected function processOrder( \Aimeos\MShop\Order\Item\Iface $order, array $params = [] )
+	{
+		$parts = \Aimeos\MShop\Order\Item\Base\Base::PARTS_SERVICE
+			| \Aimeos\MShop\Order\Item\Base\Base::PARTS_PRODUCT
+			| \Aimeos\MShop\Order\Item\Base\Base::PARTS_ADDRESS;
+
+		$base = $this->getOrderBase( $order->getBaseId(), $parts );
+		$data = $this->getData( $base, $order->getId(), $params );
+		$urls = $this->getPaymentUrls();
+
+		try
+		{
+			$provider = $this->getProvider();
+
+			if( $this->getValue( 'authorize', false ) && $provider->supportsAuthorize() )
+			{
+				$response = $provider->authorize( $data )->send();
+				$status = \Aimeos\MShop\Order\Item\Base::PAY_AUTHORIZED;
+			}
+			elseif( $this->getValue( 'apiType', '' ) === 'order' )
```

Comment:
Same order here?

Reply:
Sorry, it was a long day. Yes, same apiType in request and sync methods. I will remove the default in the other clause and switch positions here.